File: summer/WebContent/js/MS/internal/controls/ValidationRuleCollection.js
/**
 * ValidationRuleCollection
 */

define(["dojo/_base/declare", "system/Type", "objectmodel/Collection"], 
		function(declare, Type, Collection){
	var ValidationRuleCollection = declare("ValidationRuleCollection", Collection,{
		constructor:function(){
		},
		
	    /// <summary>
	    /// called by base class Collection&lt;T&gt; when an item is added to list; 
	    /// raises a CollectionChanged event to any listeners
	    /// </summary>
//	    protected override void 
		InsertItem:function(/*int*/ index, /*ValidationRule*/ item)
	    { 
	        if (item == null)
	            throw new ArgumentNullException("item"); 
	        Collection.prototype.InsertItem.call(this, index, item); 
	    },
	 
	    /// <summary>
	    /// called by base class Collection&lt;T&gt; when an item is added to list;
	    /// raises a CollectionChanged event to any listeners
	    /// </summary> 
//	    protected override void 
		SetItem:function(/*int*/ index, /*ValidationRule*/ item)
	    { 
	        if (item == null) 
	            throw new ArgumentNullException("item");
	        Collection.prototype.SetItem.call(this, index, item); 
	    }
	});
	
	ValidationRuleCollection.Type = new Type("ValidationRuleCollection", ValidationRuleCollection, [Collection.Type]);
	return ValidationRuleCollection;
});

 

File: org.summer.view.widget/src/org/summer/view/widget/controls/ValidationRuleCollection.java
package org.summer.view.widget.controls;

import org.summer.view.widget.collection.Collection;

/// <summary>
///     ValidationRulesCollection is a collection of ValidationRule
///     instances on either a Binding or a MultiBinding.  Each of the rules
///     is checked for validity on update 
/// </summary>
/*internal*/public class ValidationRuleCollection extends
		Collection<ValidationRule> {

	// -----------------------------------------------------
	//
	// Protected Methods
	//
	// -----------------------------------------------------

	// #region Protected Methods

	// / <summary>
	// / called by base class Collection&lt;T&gt; when an item is added to list;
	// / raises a CollectionChanged event to any listeners
	// / </summary>
	protected/* override */void InsertItem(int index, ValidationRule item) {
		if (item == null)
			throw new IllegalArgumentException("item");
		super.InsertItem(index, item);
	}

	// / <summary>
	// / called by base class Collection&lt;T&gt; when an item is added to list;
	// / raises a CollectionChanged event to any listeners
	// / </summary>
	protected/* override */void SetItem(int index, ValidationRule item) {
		if (item == null)
			throw new IllegalArgumentException("item");
		super.SetItem(index, item);
	}

	// #endregion Protected Methods

}